hs_core/hs_rdf.py

```diff
@@ -0,0 +1,358 @@
+
+    def get_field_terms_and_values(self, extra_ignored_fields=[]):
+        """Method that returns the field terms and field values on an object"""
+        term_values = []
+        extra_ignored_fields.extend(self.ignored_fields)
+        for field in self._meta.fields:
+            if field.name in extra_ignored_fields:
+                continue
+            field_term = self.get_field_term(field.name)
+            field_value = getattr(self, field.name)
+            if field_value and field_value is not None and field_value != 'None':
+                # urls should be a URIRef term, all others should be a Literal term
+                if isinstance(field_value, str) and field_value.startswith('http'):
+                    field_value = URIRef(field_value)
+                else:
+                    field_value = Literal(field_value)
+                term_values.append((field_term, field_value))
+        return term_values
+
+    @classmethod
+    def get_class_term(cls):
+        """
+        return a term mapped by the @rdf_terms decorator or if the deocorator is not provided, create an
+        HSTERMS.{cls.__name__}
+        """
+        return cls.rdf_term if hasattr(cls, 'rdf_term') else getattr(HSTERMS, cls.__name__)
+
+    @classmethod
+    def get_field_term(cls, field_name):
+        """
+        Given a class field_name, return a term mapped by the @rdf_terms decoroator or if the decorator is
+        not provided, create an HSTERMS.{field_name}
+        """
+        field_term_attr = field_name + '_rdf_term'
+        if hasattr(cls, field_term_attr):
+            return getattr(cls, field_term_attr)
+        else:
+            return getattr(HSTERMS, field_name)
+
+    @classmethod
+    def ingest_rdf(cls, graph, subject, content_object):
+        """Default implementation that ingests by convention"""
+        term = cls.get_class_term()
+        if not term:
+            metadata_nodes = [subject]
+        else:
+            metadata_nodes = graph.objects(subject=subject, predicate=term)
+        for metadata_node in metadata_nodes:
+            value_dict = {}
+            for field in cls._meta.fields:
+                if cls.ignored_fields and field.name in cls.ignored_fields:
+                    continue
+                field_term = cls.get_field_term(field.name)
+                val = graph.value(metadata_node, field_term)
+                if val is not None:
+                    value_dict[field.name] = str(val)
+            if value_dict:
+                cls.create(content_object=content_object, **value_dict)
+
+
+def rdf_terms(class_term, **field_terms):
+    """
+    Deocorator for mapping a class and fields to RDF terms.  The first parameter maps the class to your specified
+    term.  field_terms are key/values with the key matching a class's field and the value being the mapped term.
+
+    :raises ValidationError when a field_term key does not match a field on the class"""
```

Comment:
I like this nice rdf_terms decorator implementation. A small comment for the typo ```Deocorator``` in the comment that appear several times in the code, which is a common typo made by fast typing.

Reply:
typo fixed